src/target/riscv_debug.c

```diff
@@ -523,7 +523,7 @@ static uint32_t riscv_hart_discover_isa(riscv_hart_s *const hart)
 	do {
 		DEBUG_INFO("Attempting %u-bit read on misa\n", hart->access_width);
 		/* Try reading the register on the guessed width */
-		uint32_t isa_data[4] = {};
+		uint32_t isa_data[4] = {0, 0, 0, 0};
```

Comment:
Ah, our bad! These are missing `U` on the end of each 0 to match the data type signedness, and you can drop all bar one leaving just `{0U}` if you want (C11 says the implied members are all 0's).